clippy_lints/src/contains_for_slice.rs

```diff
@@ -0,0 +1,65 @@
+use crate::methods::method_call;
+use clippy_utils::diagnostics::span_lint;
+use rustc_hir::Expr;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{self};
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `iter().any()` on slices of `u8` or `i8` and suggests using `contains()` instead.
+    ///
+    /// ### Why is this bad?
+    /// `iter().any()` on slices of `u8` or `i8` is optimized to use `memchr`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// fn foo(values: &[u8]) -> bool {
+    ///    values.iter().any(|&v| v == 10)
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// fn foo(values: &[u8]) -> bool {
+    ///    values.contains(&10)
+    /// }
+    /// ```
+    #[clippy::version = "1.85.0"]
+    pub CONTAINS_FOR_SLICE,
+    perf,
+    "using `contains()` instead of `iter().any()` on u8/i8 slices is more efficient"
+}
+
+declare_lint_pass!(ContainsForSlice => [CONTAINS_FOR_SLICE]);
+
+impl LateLintPass<'_> for ContainsForSlice {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &Expr<'_>) {
+        if !expr.span.from_expansion()
+            // any()
+            && let Some((name, recv, _, _, _)) = method_call(expr)
+            && name == "any"
+            // iter()
+            && let Some((name, recv, _, _, _)) = method_call(recv)
+            && name == "iter"
+        {
+            // check if the receiver is a u8/i8 slice
+            let ref_type = cx.typeck_results().expr_ty(recv);
+
+            match ref_type.kind() {
+                ty::Ref(_, inner_type, _)
+                    if inner_type.is_slice()
+                        && let ty::Slice(slice_type) = inner_type.kind()
+                        && (slice_type.to_string() == "u8" || slice_type.to_string() == "i8") =>
+                {
+                    span_lint(
```

Comment:
I'm wondering if this should suggest the modified code using `span_lint_and_sugg` too...